packages/core/src/agent/Dispatcher.ts

```diff
@@ -64,7 +65,7 @@ class Dispatcher {
     await next()
   }
 
-  public async dispatch(messageContext: InboundMessageContext): Promise<void> {
+  public async dispatch(messageContext: InboundMessageContext, encryptedMessage?: EncryptedMessage): Promise<void> {
```

Comment:
What if we add `encryptedMessage` to the `InboundMessageContext` ( as it's related to the incoming message)